integrations/telegram/src/misc/utils.ts

```diff
@@ -0,0 +1,64 @@
+import { axios } from '@botpress/client'
+import { PICTURE_LIMIT_SIZE } from 'src/const'
+import { Telegraf } from 'telegraf'
+import { PhotoSize, User } from 'telegraf/typings/core/types/typegram'
+
+export const getUserNameFromTelegramUser = (telegramUser: User) => {
+  if (telegramUser.last_name) {
+    return `${telegramUser.first_name} ${telegramUser.last_name}`
+  } else if (telegramUser.username) {
+    return `${telegramUser.first_name} (username: ${telegramUser.username})`
+  }
+  return `${telegramUser.first_name} (id: ${telegramUser.id})`
+}
+
+const getDataUriFromImgHref = async (imgHref: string): Promise<string> => {
+  const fileExtension = imgHref.substring(imgHref.lastIndexOf('.') + 1)
+
+  const { data } = await axios.default.get(imgHref, { responseType: 'arraybuffer' })
+
+  const base64File = Buffer.from(data, 'binary').toString('base64')
+
+  return `data:image/${fileExtension};base64,${base64File}`
```

Comment:
issue: We should define a function with a `switch` depending on the extension that returns the proper MIME type (or alternatively use an existing NPM package, but I think it's better that we minimize dependencies for integrations as much as possible) because using the file extension directly isn't a reliable approach, as for example the MIME type for JPEG types is `image/jpeg` but these files could also come with a `.jpg` extension rather than `.jpeg` which won't work. It also applies to other image formats like SVGs (`image/svg+xml` is the MIME type) although this format isn't relevant for profile pictures, but the same principle applies 🙂

Reply:
you are right 👍 changing this